lib/everest.cpp

```diff
@@ -252,6 +255,22 @@ void Everest::register_on_ready_handler(const std::function<void()>& handler) {
     this->on_ready = std::make_unique<std::function<void()>>(handler);
 }
 
+std::optional<ModuleTierMappings> Everest::get_3_tier_model_mapping() {
+    return this->module_tier_mappings;
+}
+
+std::optional<Mapping> Everest::get_3_tier_model_mapping(const std::string& impl_id) {
+    if (not this->module_tier_mappings.has_value()) {
+        return std::nullopt;
+    }
+    auto& mapping = this->module_tier_mappings.value();
+    if (mapping.implementations.find(impl_id) == mapping.implementations.end()) {
+        // if no specific implementation mapping is given, use the module mapping
+        return mapping.module;
+    }
+    return mapping.implementations.at(impl_id);
```

Comment:
Same story with the iterator as above.  Function should be const and could also return by `const ref&` (for this you would need to have a `static std::optional<Mapping> default_value {std::nullopt}` which you would return, otherwise you directly return the references instead of unecessary copying).